### ISSUE TITLE
Image field doesn't reset itself when attachment is deleted from media library

### ISSUE THREAD

**Issue by jan-herman** [AdvancedCustomFields/acf#292]
When image selected via image field is deleted from media library this field appears to be empty but keeps the original value even after post update.

Expected bahavior:
Preferably display some kind of error message and allow user to remove the non-existent image. Alternatively display empty field but delete the original value when post is updated.

**Comment by elliotcondon** [comment]
Hi @jan-herman 

Thanks for raising the issue. You are correct in identifying that there is currently no "listener" for when a media item is deleted from the library, in regards to updating/removing saved custom field values.

Although technically possible, this would prove to be quite a potential performance issue to listen for, and update all occurrences of existing values for all object types (users, posts, terms, comments, options pages, blocks, etc).

I'm not sure there is an easy path to fixing this, but will be sure to consider it in the future :)

**Comment by jan-herman** [comment]
Hello @elliotcondon 
Thanks for such a quick reaction:)

But there already has to be some sort of detection that the image doesn't exist (client side I'm guessing) - field automatically returns to its defult state. Maybe the solution would be to also delete the value from the hidden field so the next time user saves the post it will be deleted from the database?

Or even better, leave it "broken" (do not reset the field) so it's easy to identify the problem.

Right now there is no way to tell what is causing a problem and the only way to resolve it is to select another image, remove it and then save the post.

**Comment by elliotcondon** [comment]
Yes. There is some sort of detection in place. This is run during the fields `render` function. The Image field attempts to load the selected attachment, and if that fails, no image will be displayed.
- This provides compatibility for fields without values, and also for values that are no longer valid attachments.

Thanks for raising the problem this occurs for not clearing the value. This is something we hadn't yet thought of. Currently the field will continue to save the value even if that attachment no longer exists. 

I'm happy to add this to our to-do list, and remove the hidden input value for this scenario. This would allow you to save an empty value for the page/post without the need to add/remove an image first.

We could take this one step further and delete the meta value as you mention, but I am always wary of deleteing data. We wouldn't like for a client's image field value to be deleted due to some other PHP error / conflict preventing the attachment from being loaded.

**Comment by jan-herman** [comment]
Great. Thank you:)

I belive that this relatively simple fix will save some headaches. I'm sure I'm not the only one who doesn't bother to check if the image actually exists and rely solely on get_field() value in their conditional statements:D

**Comment by elliotcondon** [comment]
Hi @jan-herman 

Yep. i agree! Thanks again for bringing this to our attention :)

**Comment by elliotcondon** [comment]
Hi @jan-herman Good news. You can expect to see this fixed in the next version (5.8.10) due for release next week :)

**Comment by jan-herman** [comment]
That was quick:) 
Great work:)
